codec-socks/src/main/java/io/netty/handler/codec/socks/SocksCmdResponseDecoder.java

```diff
@@ -28,65 +28,57 @@
  */
 public class SocksCmdResponseDecoder extends ReplayingDecoder<State> {
 
-    private SocksProtocolVersion version;
-    private int fieldLength;
-    private SocksCmdStatus cmdStatus;
-    private SocksAddressType addressType;
-    private byte reserved;
-    private String host;
-    private int port;
-    private SocksResponse msg = SocksCommonUtils.UNKNOWN_SOCKS_RESPONSE;
-
     public SocksCmdResponseDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        SocksCmdStatus cmdStatus = null;
+        SocksAddressType addressType = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksProtocolVersion.valueOf(byteBuf.readByte());
+                SocksProtocolVersion version = SocksProtocolVersion.valueOf(byteBuf.readByte());
                 if (version != SocksProtocolVersion.SOCKS5) {
                     break;
                 }
                 checkpoint(State.READ_CMD_HEADER);
             }
             case READ_CMD_HEADER: {
                 cmdStatus = SocksCmdStatus.valueOf(byteBuf.readByte());
-                reserved = byteBuf.readByte();
+                byteBuf.skipBytes(1); // reserved
                 addressType = SocksAddressType.valueOf(byteBuf.readByte());
                 checkpoint(State.READ_CMD_ADDRESS);
             }
             case READ_CMD_ADDRESS: {
                 switch (addressType) {
                     case IPv4: {
-                        host = SocksCommonUtils.intToIp(byteBuf.readInt());
-                        port = byteBuf.readUnsignedShort();
-                        msg = new SocksCmdResponse(cmdStatus, addressType, host, port);
+                        String host = SocksCommonUtils.intToIp(byteBuf.readInt());
+                        int port = byteBuf.readUnsignedShort();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)